### PR TITLE
180567810: validate logos in healthcerts submitted by clinics/providers

### PR DESCRIPTION
**Pull request by HJunyuan**
**What does this PR do?**
- Ensure provided logo is either in base64 string or HTTPS direct link (i.e. `/(data:image\/(png|jpg|jpeg);base64,.*)|(https:\/\/.*(.png|.jpg.jpeg))/`)
- Use `file-type` library to ensure file type signature
- [TEMPORARILY DISABLED] If logo is base64 string, ensure it is <= 20 * 1024 bytes (20KB)

### CODE REVIEW

File: src/functionHandlers/notarisePdt/validateInputs/mock_image.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
...
+}
```
Comment (HJunyuan) [inline src/functionHandlers/notarisePdt/validateInputs/mock_image.json:3]: Anyone has any ideas on how to programmatically create a base64 image string (of a specified size)? So we can remove this file.

File: src/functionHandlers/notarisePdt/validateInputs/validateDocument.ts [hunk 2]
```diff
@@ -168,4 +168,31 @@ export const validateV2Document = async (
   const validDomain = await isAuthorizedIssuer(issuerDomain, whitelistType);
   if (!validDomain)
     throw new UnrecognisedClinicError(issuerDomain, JSON.stringify(data.type));
+
+  /* 4. Validate logo (<=20KB base64 or https:// */
+  if (data.logo) {
+    const VALID_LOGO_PATTERN =
+      // Either base64 string or https URL in .png | .jpg | .jpeg format
+      /(data:image\/(png|jpg|jpeg);base64,.*)|(https:\/\/.*(.png|.jpg.jpeg))/;
```
Comment (HJunyuan) [inline src/functionHandlers/notarisePdt/validateInputs/validateDocument.ts:176]: Please help to double-check this regex pattern. Thanks!

**Review by eugbyte — CHANGES_REQUESTED** [review]

File: src/functionHandlers/notarisePdt/validateInputs/validateDocument.ts (hunk 2 above)
Comment (HJunyuan) [reply src/functionHandlers/notarisePdt/validateInputs/validateDocument.ts:176]: Good catch! Shall I do:
```suggestion
      /(data:image\/(png|jpg|jpeg);base64,.*)|(https:\/\/.*.(png|jpg|jpeg))/;
```

Comment (eugbyte) [reply src/functionHandlers/notarisePdt/validateInputs/validateDocument.ts:176]: i think `(.png|.jpg.jpeg))` better, as `.` itself can be interpreted as any single character. first one also looks clearer

Comment (HJunyuan) [reply src/functionHandlers/notarisePdt/validateInputs/validateDocument.ts:176]: Oh yes, good point. Actually I would still have to escape the `.` in all instances:
```suggestion
      /(data:image\/(png|jpg|jpeg);base64,.*)|(https:\/\/.*[.](png|jpg|jpeg))/;
```

**Review by zawmyolatt — APPROVED** [review]

**Review by eugbyte — APPROVED** [review]

**Pull request merged**